great_expectations/datasource/new_datasource.py

```diff
@@ -204,6 +208,35 @@ def get_batch_list_from_batch_request(
                 batches.append(new_batch)
             return batches
 
+    def _update_missing_data_asset_name(
+        self, batch_request: Union[BatchRequest, RuntimeBatchRequest]
+    ) -> None:
+        # if the data_asset_name is missing from a configured data connector config, add it as a table_name
+        if not self.config["data_connectors"][batch_request.data_connector_name][
+            "class_name"
+        ].startswith("Inferred"):
```

Comment:
Would love to hear any ideas on if there is a better way to do this.

Reply:
@NathanFarmer Is there a JIRA item defining these requirements?  Please share it via Slack (i.e., not in GitHub).  Thanks!

Reply:
✅